strictdoc/core/file_traceability_index.py

```diff
@@ -612,8 +612,9 @@ def marker_comparator(marker):
 
                 # validate here, SDocNode.relations doesn't track marker roles
                 node = traceability_index.get_node_by_uid(req_uid_)
-                assert node.document and node.document.grammar
-                grammar_element = node.document.grammar.elements_by_type[
+                document = node.get_document()
+                assert document and document.grammar
```

Comment:
Here two things:

1) Let's do two separate asserts, one for document and one for document.grammar.
2) Let's do a full version, i.e., `assert document is not None` and `assert document.grammar is not None`.

(asserting on two different objects will result in an ambiguous error message in the future
the `is not None` is what is the current convention in the codebase)
)